moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1949,8 +1949,19 @@ double RobotState::computeCartesianPath(const JointModelGroup* group, std::vecto
                                         const GroupStateValidityCallbackFn& validCallback,
                                         const kinematics::KinematicsQueryOptions& options)
 {
+  // Check whether first waypoint is identical to current pose
+  // As the current pose is set as the first trajectory waypoint anyway, this might cause issues with controllers
+  // driving towards the same target in zero time (https://github.com/ros-planning/moveit_tutorials/pull/148)
+  const Eigen::Affine3d& current_pose = getGlobalLinkTransform(link);
+  std::size_t start_index = 0;
+  if (waypoints.size() && waypoints[0].isApprox(current_pose))
+  {
+    start_index = 1;
+    ROS_WARN_NAMED("robot_state", "computeCartesianPath(): The starting point shouldn't be part of waypoints");
```

Comment:
I'm not sure if this should be a warning. There is nothing fundamentally wrong with planning a cartesian path with identical waypoints one after the other. Furthermore, this only addresses a single corner case. This same issue arrises if I call `computeCartesianPath` with `Eigen::Affine3d& target`and the target is the current pose. The same issue arrises if call `computeCartesianPath` with `EigenSTL::vector_Affine3d& waypoints` and two sequential waypoints (let's say index 2 and 3) are identical.

Reply:
That's what I pointed out [here](https://github.com/ros-planning/moveit/pull/737#pullrequestreview-88497260) already. However, @davetcoleman [argued to provide this warning anyway](https://github.com/ros-planning/moveit/pull/737#issuecomment-429652014).

Reply:
I agree that this should be a post-processing step. Personally, I am not convinced that the best way to educate users about the nuances of MoveIt! is to spam them with warnings. See my comment [here](https://github.com/ros-planning/moveit/pull/737/files#r225659287).

> I think the best thing to do would be to add a `removeDuplicateWaypoints` method that takes in a `traj` such as the one generated by `computeCartesianPath`. I would prefer if the helper function was called by `computeCartesianPath` before it returns

Reply:
> Personally, I am not convinced that the best way to educate users about the nuances of MoveIt! is to spam them with warnings.

Post processing as in remove the duplicate waypoint? Removal from a vector is very inefficient and not something that should occur in this low level a library:

> Because vectors use an array as their underlying storage, erasing elements in positions other than the vector end causes the container to relocate all the elements after the segment erased to their new positions. This is generally an inefficient operation compared to the one performed for the same operation by other kinds of sequence containers (such as list or forward_list).

Rather than compensating for bad user input forever, it seems they should be informed of their mistake so the developer can fix the higher level input source.

Imagine running this Cartesian solver in a high speed loop and being surprised to learn your trajectory is having to reshuffle the vector each call simply because you didn't know not to include the current state as the starting point? You would be annoyed this is doing this under the hood and slowing things down.

If you'd prefer an exception over a warning, that's ok with me :)